process/smartContract/processorV2/processV2.go

```diff
@@ -0,0 +1,2642 @@
+	}
+
+	return relayedSCR.OriginalTxHash
+}
+
+// ProcessIfError creates a smart contract result, consumes the gas and returns the value to the user
+func (sc *scProcessor) ProcessIfError(
+	acntSnd state.UserAccountHandler,
+	txHash []byte,
+	tx data.TransactionHandler,
+	returnCode string,
+	returnMessage []byte,
+	snapshot int,
+	gasLocked uint64,
+) error {
+	return sc.processIfErrorWithAddedLogs(acntSnd, txHash, tx, returnCode, returnMessage, snapshot, gasLocked, nil)
+}
+
+func (sc *scProcessor) processIfErrorWithAddedLogs(acntSnd state.UserAccountHandler,
+	txHash []byte,
+	tx data.TransactionHandler,
+	returnCode string,
+	returnMessage []byte,
+	snapshot int,
+	gasLocked uint64,
+	internalVMLogs []*vmcommon.LogEntry,
+) error {
+	sc.vmOutputCacher.Put(txHash, &vmcommon.VMOutput{
+		ReturnCode:    vmcommon.SimulateFailed,
+		ReturnMessage: string(returnMessage),
+	}, 0)
+
+	err := sc.accounts.RevertToSnapshot(snapshot)
+	if err != nil {
+		log.Warn("revert to snapshot", "error", err.Error())
+		return err
+	}
+
+	if len(returnMessage) == 0 {
+		returnMessage = []byte(returnCode)
+	}
+
+	acntSnd, err = sc.reloadLocalAccount(acntSnd)
+	if err != nil {
+		return err
+	}
+
+	sc.setEmptyRoothashOnErrorIfSaveKeyValue(tx, acntSnd)
+
+	scrIfError, consumedFee := sc.createSCRsWhenError(acntSnd, txHash, tx, returnCode, returnMessage, gasLocked)
+	err = sc.addBackTxValues(acntSnd, scrIfError, tx)
+	if err != nil {
+		return err
+	}
+
+	userErrorLog := createNewLogFromSCRIfError(scrIfError)
+
+	if !sc.isInformativeTxHandler(scrIfError) {
+		err = sc.scrForwarder.AddIntermediateTransactions([]data.TransactionHandler{scrIfError})
+		if err != nil {
+			return err
+		}
+	}
+
+	relayerLog, err := sc.processForRelayerWhenError(tx, txHash, returnMessage)
+	if err != nil {
+		return err
+	}
+
+	processIfErrorLogs := make([]*vmcommon.LogEntry, 0)
+	processIfErrorLogs = append(processIfErrorLogs, userErrorLog)
+	if relayerLog != nil {
+		processIfErrorLogs = append(processIfErrorLogs, relayerLog)
+	}
+	if len(internalVMLogs) > 0 {
+		processIfErrorLogs = append(processIfErrorLogs, internalVMLogs...)
+	}
+
+	logsTxHash := sc.getOriginalTxHashIfIntraShardRelayedSCR(tx, txHash)
+	ignorableError := sc.txLogsProcessor.SaveLog(logsTxHash, tx, processIfErrorLogs)
+	if ignorableError != nil {
+		log.Debug("scProcessor.ProcessIfError() txLogsProcessor.SaveLog()", "error", ignorableError.Error())
+	}
+
+	txType, _ := sc.txTypeHandler.ComputeTransactionType(tx)
+	isCrossShardMoveBalance := txType == process.MoveBalance && check.IfNil(acntSnd)
+	if isCrossShardMoveBalance {
+		// move balance was already consumed in sender shard
+		return nil
+	}
+
+	sc.txFeeHandler.ProcessTransactionFee(consumedFee, big.NewInt(0), txHash)
+
+	err = sc.blockChainHook.SaveNFTMetaDataToSystemAccount(tx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (sc *scProcessor) setEmptyRoothashOnErrorIfSaveKeyValue(tx data.TransactionHandler, account state.UserAccountHandler) {
```

Comment:
leave a TODO here - to delete this function and actually resolve the issue in the built in function of saveKeyValue.

Reply:
fixed